src/NzbDrone.Core/Download/Clients/FreeboxDownload/FreeboxDownloadProxy.cs

```diff
@@ -0,0 +1,277 @@
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Security.Cryptography;
+using NLog;
+using NzbDrone.Common.Cache;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Common.Http;
+using NzbDrone.Common.Serializer;
+using NzbDrone.Core.Download.Clients.FreeboxDownload.Responses;
+
+namespace NzbDrone.Core.Download.Clients.FreeboxDownload
+{
+    public interface IFreeboxDownloadProxy
+    {
+        void AuthVerify(FreeboxDownloadSettings settings);
+        string AddTaskFromUrl(string url, string directory, bool addPaused, bool addFirst, double? seedRatio, FreeboxDownloadSettings settings);
+        string AddTaskFromFile(string fileName, byte[] fileContent, string directory, bool addPaused, bool addFirst, double? seedRatio, FreeboxDownloadSettings settings);
+        void DeleteTask(string id, bool deleteData, FreeboxDownloadSettings settings);
+        FreeboxDownloadConfiguration GetDownloadConfiguration(FreeboxDownloadSettings settings);
+        List<FreeboxDownloadTask> GetTasks(FreeboxDownloadSettings settings);
+    }
+
+    public class FreeboxDownloadProxy : IFreeboxDownloadProxy
+    {
+        private readonly IHttpClient _httpClient;
+        private readonly Logger _logger;
+        private ICached<string> _authSessionTokenCache;
+
+        public FreeboxDownloadProxy(ICacheManager cacheManager, IHttpClient httpClient, Logger logger)
+        {
+            _httpClient = httpClient;
+            _logger = logger;
+            _authSessionTokenCache = cacheManager.GetCache<string>(GetType(), "authSessionToken");
+        }
+
+        public void AuthVerify(FreeboxDownloadSettings settings)
```

Comment:
I think `Authenticate` would be a better name.

Reply:
Done!